uber/utils.py

```diff
@@ -262,17 +262,30 @@ def charge_cc(self, token):
             )
         except stripe.CardError as e:
             return 'Your card was declined with the following error from our processor: ' + str(e)
-        except stripe.error.InvalidRequestError as e:
-            log.error('Invalid source object provided', exc_info=True)
-            send_email(c.ADMIN_EMAIL, [c.ADMIN_EMAIL, 'dom@magfest.org'], 'MAGFest Stripe invalid request error',
-                       'Got an error while calling charge_cc(self, token={!r}):\n{}'
-                       .format(token, traceback.format_exc()))
-            return 'An invalid source object was provided: ' + str(e)
         except stripe.StripeError as e:
-            log.error('unexpected stripe error', exc_info=True)
+            error_txt = 'Got an error while calling charge_cc(self, token={!r})'.format(token)
+            report_critical_exception(msg=error_txt, subject='ERROR: MAGFest Stripe invalid request error')
             return 'An unexpected problem occured while processing your card: ' + str(e)
 
 
+def report_critical_exception(msg, subject="Critical Error"):
+    """
+    Report an exception to the loggers with as much context (request params/etc) as possible, and send an email.
+
+    Call this function when you really want to make some noise about something going really badly wrong.
+
+    :param msg: message to prepend to output
+    :param subject: optional: subject for emails going out
+    """
+
+    # log with lots of cherrypy context in here
+    uber.server.log_exception_with_verbose_context(msg)
+
+    # also attempt to email the admins
+    # TODO: Don't hardcode emails here.
+    send_email(c.ADMIN_EMAIL, [c.ADMIN_EMAIL, 'dom@magfest.org'], subject, msg + '\n{}'.format(traceback.format_exc()))
```

Comment:
i left this harcocded email in cause it was already there and I don't have time to puppetize this option right now